infra/Pos/Recovery/Info.hs

```diff
@@ -75,25 +76,37 @@ recoveryInProgress =
         SSDoingRecovery -> True
         _ -> False
 
+-- | Get sync status using K as lagBehind param.
+getSyncStatusK :: (MonadRecoveryInfo m, HasConfiguration) => m SyncStatus
+getSyncStatusK = getSyncStatus slotSecurityParam
+  where
+    -- It's actually questionable which value to use here. The less it
+    -- is, the stricter is the condition to do some
+    -- work. 'slotSecurityParam' is reasonable, but maybe we should use
+    -- something smaller.
+    lagBehindParam :: SlotCount
+    lagBehindParam = slotSecurityParam
+
 -- | This is a helper function which runs given action only if we are
 -- kinda synchronized with the network.  It is useful for workers
 -- which shouldn't do anything while we are not synchronized.
 recoveryCommGuard
     :: (MonadRecoveryInfo m, WithLogger m, HasConfiguration)
     => Text -> m () -> m ()
 recoveryCommGuard actionName action =
-    getSyncStatus lagBehindParam >>= \case
+    getSyncStatusK >>= \case
         SSKindaSynced -> action
-        status -> onIgnore status
-  where
-    -- It's actually questionable which value to use here. The less it
-    -- is, the stricter is the condition to do some
-    -- work. 'slotSecurityParam' is reasonable, but maybe we should use
-    -- something smaller.
-    lagBehindParam :: SlotCount
-    lagBehindParam = slotSecurityParam
-    onIgnore status =
-        logDebug $
-        sformat
-            ("recoveryCommGuard: we are skipping action '"%stext%
-             "', because "%build) actionName status
+        status ->
+            logDebug $
+            sformat ("recoveryCommGuard: we are skipping action '"%stext%
+                     "', because "%build) actionName status
+
+-- | This function checks that last known block is more than K slots
+-- away from the current slot, or current slot isn't known. It also
+-- returns False when we're actually doing recovery. So basically it
+-- returns true if we actually need to ask for tips right now.
+needTriggerRecovery :: (MonadRecoveryInfo m, HasConfiguration) => m Bool
+needTriggerRecovery = getSyncStatusK <&> \case
+    SSKindaSynced   -> False
+    SSDoingRecovery -> False
```

Comment:
What about `SSInFuture`? We likely don't need recovery in this case.

Reply:
What do you think?

Reply:
Ah, ok, noticed that in the second commit. Perhaps it'd be better to amend this one instead, but whateva.